src/include/sof/audio/buffer.h

```diff
@@ -65,6 +67,13 @@ struct comp_buffer {
 	void (*cb)(void *data, uint32_t bytes);
 	void *cb_data;
 	int cb_type;
+
+	/* runtime stream params */
+	uint32_t frame_fmt;	/**< enum sof_ipc_frame */
+	uint32_t buffer_fmt;	/**< enum sof_ipc_buffer_format */
+	uint32_t rate;
+	uint16_t channels;
+	uint16_t chmap[SOF_IPC_MAX_CHANNELS];	/**< channel map - SOF_CHMAP_ */
```

Comment:
Do *all* of these really have to be fixed-width fields? E.g. rate? Wouldn't (unsigned) int do for it? Or channels? If formats are really those enums, why not also declare them as such?

Reply:
@lyakh 
I've just moved it to other struct. I do not mind changing these types to enum. 
@lgirdwood 
Is there a specific reason why we used fixed-width fields instead of e.g. enums?